server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -450,9 +489,15 @@ public Indexer(List<String> partitionValues,
             if (targetColumns.contains(ref) || ref.granularity() == RowGranularity.PARTITION) {
                 continue;
             }
+            // never re-generate default root columns for UPDATE
+            // default sub-columns may need to be re-generated, i.e. update t set o={}, where o has default sub-cols
+            boolean isUpdate = updateColumns != null && updateColumns.length > 0 && targetColumns.isEmpty();
```

Comment:
Could move the `isUpdate` assignment out of the loop to do the evaluation only once